### PR TITLE
ZH-trans: add cloudstack.md

### PR DESCRIPTION
**Pull request by xichengliudui**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Please delete this note before submitting the pull request.
>
> For 1.14 Features: set Milestone to 1.14 and Base Branch to dev-1.14
> 
> For Chinese localization, base branch to release-1.12
>
> For Korean Localization: set Base Branch to dev-1.13-ko.<latest team milestone>
>
> Help editing and submitting pull requests:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> Help choosing which branch to use:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

File: content/zh/docs/setup/on-premises-vm/cloudstack.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+{{% capture overview %}}
+
+<!--
+[CloudStack](https://cloudstack.apache.org/) is a software to build public and private clouds based on hardware virtualization principles (traditional IaaS). 
+To deploy Kubernetes on CloudStack there are several possibilities depending on the Cloud being used and what images are made available. 
+CloudStack also has a vagrant plugin available, hence Vagrant could be used to deploy Kubernetes either using the existing shell provisioner or using new Salt based recipes.
+-->
+
+[CloudStack](https://cloudstack.apache.org/) 是一种基于硬件虚拟化原则(传统 IaaS) 构建公有云和私有云的软件。要在 CloudStack 上部署 Kubernetes，
+有几种可能性取决于正在使用的云以及可用的图像。CloudStack 还提供了一个 vagrant 插件，因此 vagrant 可以使用现有的 shell 供应程序或新的基于 Salt 的方法来部署 Kubernetes。
```
Comment (chenrui333) [inline content/zh/docs/setup/on-premises-vm/cloudstack.md]: image, 镜像 😄
```suggestion
有几种可能性取决于正在使用的云以及可用的镜像。CloudStack 还提供了一个 vagrant 插件，因此 vagrant 可以使用现有的 shell 创建程序，或新的基于 Salt 的方法来部署 Kubernetes。
```

Comment (xichengliudui) [reply content/zh/docs/setup/on-premises-vm/cloudstack.md]: 完成！

File: content/zh/docs/setup/on-premises-vm/cloudstack.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+{{% capture overview %}}
+
+<!--
+[CloudStack](https://cloudstack.apache.org/) is a software to build public and private clouds based on hardware virtualization principles (traditional IaaS). 
+To deploy Kubernetes on CloudStack there are several possibilities depending on the Cloud being used and what images are made available. 
+CloudStack also has a vagrant plugin available, hence Vagrant could be used to deploy Kubernetes either using the existing shell provisioner or using new Salt based recipes.
+-->
+
+[CloudStack](https://cloudstack.apache.org/) 是一种基于硬件虚拟化原则(传统 IaaS) 构建公有云和私有云的软件。要在 CloudStack 上部署 Kubernetes，
```
Comment (chenrui333) [inline content/zh/docs/setup/on-premises-vm/cloudstack.md]:
```suggestion
[CloudStack](https://cloudstack.apache.org/) 是一种基于硬件虚拟化原则(传统 IaaS 概念)用来构建公有云和私有云的软件。要在 CloudStack 上部署 Kubernetes，
```

File: content/zh/docs/setup/on-premises-vm/cloudstack.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+有几种可能性取决于正在使用的云以及可用的图像。CloudStack 还提供了一个 vagrant 插件，因此 vagrant 可以使用现有的 shell 供应程序或新的基于 Salt 的方法来部署 Kubernetes。
+
+<!--
+[CoreOS](http://coreos.com) templates for CloudStack are built [nightly](http://stable.release.core-os.net/amd64-usr/current/). 
+CloudStack operators need to [register](http://docs.cloudstack.apache.org/projects/cloudstack-administration/en/latest/templates.html) this template in their cloud before proceeding with these Kubernetes deployment instructions.
+-->
+[CoreOS](http://coreos.com) 为 CloudStack [每夜](http://stable.release.core-os.net/amd64-usr/current/)构建模板。在继续执行这些 Kubernetes 部署指令之前，CloudStack 操作人员需要在他们的云中 [注册](http://docs.cloudstack.apache.org/projects/cloudstack-administration/en/latest/templates.html)这个模板。
```
Comment (chenrui333) [inline content/zh/docs/setup/on-premises-vm/cloudstack.md]: operator 是 CoreOS operator framework, 这里不需要翻译。
```suggestion
CloudStack 的 [CoreOS](http://coreos.com) 模板会[每夜](http://stable.release.core-os.net/amd64-usr/current/)构建。
CloudStack operators 需要在他们的云中 [注册](http://docs.cloudstack.apache.org/projects/cloudstack-administration/en/latest/templates.html)这个模板，然后才能继续执行 Kubernetes 部署的命令。
```

File: content/zh/docs/setup/on-premises-vm/cloudstack.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+<!--
+This guide uses a single [Ansible playbook](https://github.com/apachecloudstack/k8s), which is completely automated and can deploy Kubernetes on a CloudStack based Cloud using CoreOS images. The playbook, creates an ssh key pair, creates a security group and associated rules and finally starts coreOS instances configured via cloud-init.
+-->
+
+本指南使用一个 [Ansible 副本](https://github.com/apachecloudstack/k8s)，它是完全自动化的，可以使用 CoreOS 镜像在基于 CloudStack 的云上部署 Kubernetes。副本创建 ssh 密钥对、创建安全组和相关规则，最后启动通过 cloud-init 配置的 coreOS 实例。
```
Comment (chenrui333) [inline content/zh/docs/setup/on-premises-vm/cloudstack.md]: playbook 是Ansible 的术语，也可以考虑不用翻译。
https://getansible.com/advance/playbook/playbook

```suggestion
本指南只用到一个 [Ansible playbook](https://github.com/apachecloudstack/k8s)，完全自动化，可以使用 CoreOS 镜像在基于 CloudStack 的云上部署 Kubernetes。playbook 会创建 ssh 密钥对、创建安全组和相关规则，最后启动通过 cloud-init 配置的 CoreOS 实例。
```

File: content/zh/docs/setup/on-premises-vm/cloudstack.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+<!--
+We need to use the http POST method to pass the _large_ userdata to the coreOS instances.
+-->
+我们需要使用 http POST 方法将 _large_ userdata 传递给 coreOS 实例。
+
+<!--
+### Clone the playbook
+-->
+
+### 克隆的副本
```
Comment (chenrui333) [inline content/zh/docs/setup/on-premises-vm/cloudstack.md]:
```suggestion
### 复制 playbook
```

File: content/zh/docs/setup/on-premises-vm/cloudstack.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+<!--
+### Create a Kubernetes cluster
+-->
+
+### 创建一个 Kubernetes 集群
+
+<!--
+You simply need to run the playbook.
+-->
+你只需要运行副本。
```
Comment (chenrui333) [inline content/zh/docs/setup/on-premises-vm/cloudstack.md]:
```suggestion
你只需要运行 playbook 即可。
```

File: content/zh/docs/setup/on-premises-vm/cloudstack.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+## 支持级别
+
+<!--
+IaaS Provider        | Config. Mgmt | OS     | Networking  | Docs                                              | Conforms | Support Level
+-------------------- | ------------ | ------ | ----------  | ---------------------------------------------     | ---------| ----------------------------
+CloudStack           | Ansible      | CoreOS | flannel     | [docs](/docs/setup/on-premises-vm/cloudstack/)                             |          | Community ([@Guiques](https://github.com/ltupin/))
+-->
+
+IaaS Provider        | Config. Mgmt | OS     | Networking  | Docs                                              | Conforms | Support Level
```
Comment (chenrui333) [inline content/zh/docs/setup/on-premises-vm/cloudstack.md]: table 里面的标题也可以翻译一下
```suggestion
IaaS 供应商        | 配置管理 | 操作系统     | 网络  | 文档                                              | 规范 | 支持级别
```

File: content/zh/docs/setup/on-premises-vm/cloudstack.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+-------------------- | ------------ | ------ | ----------  | ---------------------------------------------     | ---------| ----------------------------
+CloudStack           | Ansible      | CoreOS | flannel     | [文档](/docs/setup/on-premises-vm/cloudstack/)                             |          | Community ([@Guiques](https://github.com/ltupin/))
```
Comment (chenrui333) [inline content/zh/docs/setup/on-premises-vm/cloudstack.md]:
```suggestion
CloudStack           | Ansible      | CoreOS | flannel     | [文档](/docs/setup/on-premises-vm/cloudstack/)                             |          | 社区 ([@Guiques](https://github.com/ltupin/))
```

**Comment by xichengliudui** [comment]
一段时间没翻译了，下面我会减少这些问题。

**Comment by chenrui333** [comment]
没事的，reviewer 就是干这事的 😄

**Comment by xichengliudui** [comment]
已更新。

**Pull request merged**